eth/gasprice/gasprice.go

```diff
@@ -333,7 +336,14 @@ func (oracle *Oracle) suggestDynamicFees(ctx context.Context) (*big.Int, *big.In
 		tipResults     []*big.Int
 		baseFeeResults []*big.Int
 	)
-
+	// if the fee config has changed in this block, use the new base fee
+	// if it is higher than this block's base fee.
+	if feeLastChangedAt != nil && feeLastChangedAt.Uint64() == number {
+		if lastBaseFee.Cmp(feeConfig.MinBaseFee) < 0 {
+			lastBaseFee = feeConfig.MinBaseFee
+			// TODO: should we also reset lastPrice?
```

Comment:
Since `lastPrice` is derived from the tip or amount above base fee people are willing to pay, I think we could go either way on this depending on how we expect people's willingness to pay above the base fee to be impacted by the fee config change.

So I don't think we necessarily need to reset it here.

Reply:
I believe we need to handle excluding base fees that are from a different "fee config era" completely rather than only setting the `lastBaseFee` if there's an exact match.

Reply:
that is already handled in line #348

Reply:
Ah I see